src/main/java/me/moodcat/api/ChatAPI.java

```diff
@@ -0,0 +1,131 @@
+package me.moodcat.api;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import lombok.Data;
+import me.moodcat.database.controllers.ChatDAO;
+import me.moodcat.database.controllers.RoomDAO;
+import me.moodcat.database.entities.ChatMessage;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+/**
+ * API to send and check for messages in a room.
+ */
+@Singleton
+@Path("/api/rooms/{roomId}/chat")
+@Produces(MediaType.APPLICATION_JSON)
+public class ChatAPI {
+
+    private final ChatDAO chatDAO;
+
+    private final RoomDAO roomDAO;
+
+    private ArrayList<ChatMessage> messages;
```

Comment:
Point to the interface if you don't need a specific implementation. This allows to switch implementations easier.